### PR TITLE
Initializer bug with inheritance and inlining

### PR DESCRIPTION
**Pull request by benharsh**
Add test for inheriting classes and initializers that fails during inlining

### CODE REVIEW

**Comment by benharsh** [comment]
@lydia-duncan , @noakesmichael : I couldn't find an existing test case for this bug. Does it seem familiar to either of you?

File: test/classes/initializers/generics/inheritance/inheritance-inline.chpl [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  }
+}
+
+class Simple : Base {
+  proc init() {
+  }
+}
+
+var f = new Fancy();
+var s = new Simple();
```
Comment (lydia-duncan) [inline test/classes/initializers/generics/inheritance/inheritance-inline.chpl]: You'll want to delete these instances to avoid memory leaks, btw

**Comment by lydia-duncan** [comment]
That doesn't seem familiar to me at all, that's a really weird failure.

**Comment by lydia-duncan** [comment]
Oh, I think I know why.  Mike, this seems related to making the "init" function inlined when we create the "_new" function.  It probably thinks the implicit "super.init()" call is recursive, but that is still strange.

**Comment by benharsh** [comment]
OK, classes are deleted (good catch). Do you want to add this to the big list of issues, or should it be a separate one?

**Pull request merged**